### PR TITLE
Yield chart

### PR DESCRIPTION
**Pull request by katieemelianova**
Still working on this, creating a pull request now to review structure

### CODE REVIEW

File: reporting_app/util.py [hunk 1]
```diff
@@ -1,6 +1,9 @@
 import requests
 from flask import json
 from config import reporting_app_config as cfg, col_mappings
+import numpy
+import math
+import datetime
```
Comment (tcezard) [inline reporting_app/util.py]: It looks like  numpy, math, and datetime are not used here.
numpy also add a unnecessary dependency

File: reporting_app/templates/untabbed_datatables.html [hunk 2]
```diff
@@ -20,4 +20,270 @@
     {{ datatable(table) }}
 {% endif %}
 
-{% endblock %}
+
+<script type="text/javascript" src="https://www.gstatic.com/charts/loader.js"></script>
```
Comment (mwhamgenomics) [inline reporting_app/templates/untabbed_datatables.html]: This is not related to setting up datatables, so should be split off into its own template and .js script. There should then be a new 'pipeline_view' template that mixes in this and `untabbed_datatables.html`.

File: reporting_app/__init__.py [hunk 3]
```diff
@@ -32,22 +32,41 @@ def pipeline_report(pipeline_type, view_type):
...
     if view_type == 'all':
         query = rest_query(endpoint)
+        run_data = query_api('aggregate/all_runs')
+        sample_data = query_api('aggregate/samples')
+
     elif view_type in statuses:
         query = rest_query(endpoint, match={'$or': [{'proc_status': s} for s in statuses[view_type]]})
+        args = ({'$or': [{'proc_status': s} for s in statuses[view_type]]})
+        run_data = query_api('aggregate/all_runs', match=args)
+        sample_data = query_api('aggregate/samples', match=args)
```
Comment (mwhamgenomics) [inline reporting_app/__init__.py]: This results in 2 API queries, but only 1 is ever used. We should decide whether to render a sample or run view, then to API queries and render templates accordingly.
(API queries should also go through `egcg_core` once the `apache` branch is merged)

File: reporting_app/templates/untabbed_datatables.html [hunk 4]
```diff
@@ -20,4 +20,270 @@
...
+
+      return [weeks, months, aggregate_weeks, aggregate_months]
+
+}
+
+
+
+function drawCharts() {
+
+      var pipeline_type = {{ pipeline | safe }};
```
Comment (mwhamgenomics) [inline reporting_app/templates/untabbed_datatables.html]: Add quotes around `{{ pipeline }}`, then you won't need to pass this in as a list of 1 item, and you won't need to pass it through `safe`.

File: reporting_app/templates/untabbed_datatables.html [hunk 5]
```diff
@@ -20,4 +20,270 @@
...
+          interpolateNulls: false};
+          var chart2 = new google.visualization.Histogram(document.getElementById('yield_histogram'));
+          chart2.draw(histogram_data, histogram_options);
+      }
+}
+
+</script>
+</head>
+
+<body>
```
Comment (mwhamgenomics) [inline reporting_app/templates/untabbed_datatables.html]: These `div`s should be in the template's `content` block, rather than hard-coded to the `<body>`.

**Comment by tcezard** [comment]
I had a quick test and I don't think the plots are showing accurate data from the runs/samples that are extracted.
Can you double check?

File: reporting_app/__init__.py [hunk 6]
```diff
@@ -3,6 +3,8 @@
 import flask as fl
 import flask_login
 import auth
+from reporting_app.util import datatable_cfg, tab_set_cfg, chart_variables, yield_by_date, sample_sequencing_metrics, \
+    get_token
 from config import reporting_app_config as cfg
 from reporting_app.util import datatable_cfg, tab_set_cfg
```
Comment (mwhamgenomics) [inline reporting_app/__init__.py]: We've already imported these things 2 lines above, so can remove this line.

File: reporting_app/util.py [hunk 7]
```diff
@@ -33,10 +34,106 @@ def tab_set_cfg(title, tables):
...
 
 
 def snake_case(text):
     return text.lower().replace(' ', '_')
+
+
+
+
+def chart_variables(endpoint, data):
+
```
Comment (mwhamgenomics) [inline reporting_app/util.py]: Is this function used anywhere? I can see it being imported in `reporting_app.__init__`, but no usages.

File: reporting_app/static/charts.js [hunk 8]
```diff
@@ -0,0 +1,228 @@
...
+
+function unwind_samples_sequenced(sample_data){
+    var all_sample_data = [];
+    for (var e=0; e < sample_data.length; e++) {
+        d = sample_data[e];
+        run_ids = d['all_run_ids'].slice(0);
+        if (run_ids.length > 0){
+            run_ids.sort();
+            console.log(d['sample_id']);
+            console.log(run_ids)
```
Comment (mwhamgenomics) [inline reporting_app/static/charts.js]: Don't need these print statements, as far as I can see. I'd try to print as little as possible, and use `console.warn` or `console.error` when stuff goes wrong.

File: reporting_app/util.py [hunk 9]
```diff
@@ -33,10 +34,106 @@ def tab_set_cfg(title, tables):
...
+    run_date = int(run_id.split('_')[0])
+    year = int(''.join(list(str(run_date))[0:2])) + 2000
+    month = int(''.join(list(str(run_date))[2:4]))
+    day = int(''.join(list(str(run_date))[4:6]))
+    d = datetime.datetime(year, month, day)
+    date = int(time.mktime(d.timetuple())) * 1000
+    return date
+
+
+def yield_by_date(data):
```
Comment (mwhamgenomics) [inline reporting_app/util.py]: Not seeing any usages of this function, either.

File: reporting_app/util.py [hunk 10]
```diff
@@ -33,10 +34,106 @@ def tab_set_cfg(title, tables):
...
+    for d in data:
+        run_id = d.get('_id')
+        date = run_name_to_date(run_id)
+        run_yield = [d.get('yield_in_gb')]
+        date_yield.append([date, run_yield])
+    return date_yield
+
+
+def sample_sequencing_metrics(data):
+    date_samples_first_sequenced = []
```
Comment (mwhamgenomics) [inline reporting_app/util.py]: ... or this.
